### ISSUE TITLE
Oddity with autoload: files

### ISSUE THREAD

**Issue by SignpostMarv** [vimeo/psalm#127]
Currently in the process of running static analysis on a project before rewriting it- in this particular case, I've got a  class that's not namespaced & psalm 7d17a07c reports a `DuplicateClass` error.
> ERROR: DuplicateClass - src\Response.php:50 - Class Response has already been defined internally

Additionally, setting the error type to the errorLevel of "info" doesn't allow psalm to continue testing as expected.

Triggers the error:
```json
    "autoload": {
        "files": [
            "src/Response.php"
        ],
    }
```

Does not trigger the error & allows the testing to continue as expected, (but triggers a warning with `composer validate`):
```json
    "autoload": {
        "psr-4": {
            "": "src/"
        }
    },
```

**Comment by SignpostMarv** [comment]
another example of the error;

```json
{
    "autoload": {
        "psr-4": {
        }
    },
    "autoload-dev": {
        "files": [
            "./app/migrations/20170625164232_CreateTableOAuthScopes.php"
        ]
    },
    "config": {
        "classmap-authoritative": true,
        "optimize-autoloader": true,
        "sort-packages": true
    }
}
```

where `"./app/migrations/20170625164232_CreateTableOAuthScopes.php"` is a migration class for Phinx

**Comment by muglug** [comment]
Are you able to create a repo that exhibits this error? I imagine I'd be able to fix it pretty quickly then.

**Comment by SignpostMarv** [comment]
https://github.com/SignpostMarv/psalm-127-reproduction

added in a class extension for extra kicks.

**Comment by muglug** [comment]
Thanks for that repo!
